Ports/build_all.sh

```diff
@@ -58,19 +58,19 @@ for file in *; do
                 fi
             fi
             if [ "$verbose" == true ]; then
-                if ./package.sh; then
+                if ./package.sh > build-log.files; then
```

Comment:
hm, this seems generally useful. I wonder if adding `.log` to the gitignore (if it's not already there) would be better? Not sure if we have any files in-repo with that extension.  Since this is in the verbose section, it should probably `| tee build-log.files` instead to keep the output going to stdout.

That being said...

The general pattern we've been moving to has been to put the per-package-build artifacts in `Build/$ARCH/Ports/` and avoid putting any temporary files in the Ports/ directory at all though. ...which involves the `target_env` function from `.port_include.sh`. That would probably be messy to include from here. 

I wonder if modifying `.port_include.sh` with an exit handler and writing to a file such as `$PORT_BUILD_DIR/../failed-ports.log` would be a better approach? Doing this there would also allow doing weird `sh` things to dup/tee stdout/stderr to `$PORT_BUILD_DIR/last-build.log`.

https://github.com/SerenityOS/serenity/blob/b01fcccbbd43abea157143ebfa64976dc389278c/Ports/.port_include.sh#L111-L116

Reply:
There are many .log files, I'm not sure are they safe to ignore:

https://gist.github.com/xet7/2bed2bfc43f6dc864b4b2c5d9cbdd15b

Reply:
In serenity directory, .log files can be found with:
```
find . -type f -name "*.log"
```

Reply:
I needed per-port logs, because I found what dependencies were missing to build DosBox https://github.com/wekan/wekan/wiki/serenity#dosbox . I was trying to find, should those dependencies be in readme, at each port, or somewhere else.

Currently those build dependencies are distro specific at https://github.com/SerenityOS/serenity/blob/master/Documentation/BuildInstructions.md#debian--ubuntu . Those distros and operating systems could be detected in bash script, so that same install script would work in all of them

There is some OS detection code here: https://github.com/wekan/wekan/blob/master/rebuild-wekan.sh#L25

Reply:
As I also said on Discord: I'm personally against having log files in source directories; if we have them, they should go in the port build dir.

I'm not sure I understand the problem being solved though. I've never had problems finding which dependencies were missing?

Reply:
Problem is, how to install all dependencies required by all Ports? build_all.sh script shows many failed builds.

Reply:
And is there any script, that installs all dependencies, SerenityOS, and all Ports at once, on Ubuntu 23.04 ?

Reply:
@xet7 Maybe you can join us on Discord in the #ports channel, so we can discuss what you're trying to achieve? Sounds very interesting!

At any rate:

> Problem is, how to install all dependencies required by all Ports? build_all.sh script shows many failed builds.

We don't have the dependencies documented, so you'd need to look at the output of the builds to see what dependencies are missing. Redirecting the output of `build_all.sh` to log files by default seems a bit blunt to me.

> And is there any script, that installs all dependencies, SerenityOS, and all Ports at once, on Ubuntu 23.04 ?

Nope, we don't support that yet. Amongst others, the issues that we need to solve are 1) that we need to know which dependencies exactly are required at a minimum for each port and 2) what those dependencies are called on different Linux distributions (+ WSL) and MacOS and 3) whether or not these dependencies are provided through package managers, manual installs, etc.

It's not a straightforward issue to solve, so let's discuss first!

Reply:
Regarding broken Ports, it might also be helpful to double-check with https://github.com/SerenityOS/serenity/issues/18238 and post an update if anything changed.
(Ideally we would have an auto-updating table, but …)